core/app/models/spree/promotion.rb

```diff
@@ -50,7 +50,7 @@ class Promotion < Spree::Base
     self.whitelisted_ransackable_attributes = ['path', 'promotion_category_id']
 
     # temporary code. remove after the column is dropped from the db.
-    def columns
+    def self.columns
```

Comment:
This should fix the issue but what about using [ingored_columns](https://api.rubyonrails.org/classes/ActiveRecord/ModelSchema/ClassMethods.html#method-i-ignored_columns) instead?